




![Kickstarter_logo](Kickstarter_Logo.png)

Launched in 2009, **Kickstarter** is one of the world's leading crowdfunding platforms. As of December 2019, Kickstarter has received more than $4.6 billion in pledges from 17.2 million backers to fund 445,000 projects, such as films, music, stage shows, comics, journalism, video games, technology, publishing, and food-related projects. It's mission is to bring "*creative projects to life*"

Kickstarter, founded in 2009, is one particularly well-known and popular crowdfunding platform. It has an all-or-nothing funding model, whereby a project is only funded if it meets its goal amount; otherwise no money is given by backers to a project.
A huge variety of factors contribute to the success or failure of a project — in general, and also on Kickstarter. Some of these are able to be quantified or categorized, which allows for the construction of a model to attempt to predict whether a project will succeed or not.

## Objective of the Analysis 

1. Primary **business objective** is to provide creators with a recommendation on how to launch a successful Kickstarter campaign! 

2. Determine which factors decide whether or not a project will achieved is funding goal. 

### How do we define a "successful" Project? 

Success in the context of the Kickstarter Dataset is defined as achieving the funding goal.

## Project Requirements: 

1. Try different (at least 3) machine learning algorithms to check which performs best on the problem at hand
2. What would be the right performance metric- precision, recall, accuracy, F1 score, or something else? (Check TPR?)

**Hint**: Check for Data imbalance


## Description of the Dataset 

**'backers_count':** Number of folks who pledge money to join creators in bringing projects to life

**'blurb':** Description of the project / company

**'category':** Describes the topic of the project (e.g. music, fashion)

**'converted_pledged_amount':** Amount of money pledged, converted to the currency on the `current_currency`column 

**'country':** Country where the project creators originates from 

**'created_at':** Date and time when the project was initially created on Kickstarter

**'creator':** The person or team behind the project idea, working to bring it to life

**'currency':** Name of original currency 

**'currency_symbol':** corresponding currency symbol

**'currency_trailing_code':** ?

**'current_currency':** Currency after the conversion has taken place

**'deadline':** Final crowdfunding date

**'disable_communication':** whether or not a project creator is able to communicate with they backers

**'friends':** unclear, null or empty

**'fx_rate':** Foreign exchange rate between the original currency and the current currency

**'goal':** The amount of money that a creator needs to complete their project. Minimum requirement for the project to be financed

**'id':** Project ID

**'is_backing':** 

**'is_starrable':** provides the option to leave a star review

**'is_starred':** has received a star review

**'launched_at':** state and time when the project was launched for funding at Kickstarter

**'location':** Contains the town or city of the project creator

**'name':** Name of the campaign

**'permissions':** unclear; is either NA or empty in the dataset

**'photo':** contains a link and information to the projects photos

**'pledged':** Amount pledged by the contributors in the original currency 

**'profile':** Details about the projects profile including ID number and various visual settings

**'slug':** Name of the project with hyphens and lowercase letters instead of spaces and uppercase letters

**'source_url':** link to the project category on the Kickstarter website

**'spotlight':** Option to put the campaign in a spotlight via a landing page on Kickstarter after it has been successfully financed

**'staff_pick':** Whether a project was handpicked and highlighted by the Kickstarter team. These projects are displayed favorably on the Kickstarter page.

**'state':** Status of the campaign that can be classified into one of the following categories:
   * *'successful'*: project has achieved the funding goal and is neither canceled or suspended. (only classified as successful after the deadline has passed?)
   * *'failed'*: project has failed to achieve the funding goal within the deadline
   * *'live'*: Campaign are classified as live when they are still ongoing regardless of whether they have already achieved the funding goal or not
   * *'suspended'*: A project may be suspended if the Trust & Safety team uncovers evidence that it is in violation of Kickstarter's rules
   * *'canceled'*: A project may be canceled if the creator wants to make any major changes to the project, such as the funding goal or campaign duration, or likes to rework the idea and start again
   
**'state_changed_at':** Date and time when a project status was changed (e.g. from live to successful / failed)

**'static_usd_rate':** Conversion rate between the original currency and USD

**'urls':** link to the creator's campaign on Kickstarter

**'usd_pledged':** Pledged amount converted to USD done by Kickstarter

**'usd_type':** unclear, classifies either as domestic or international 

##  Import Libaries

In [63]:
import pandas as pd
import numpy as np
import glob
import matplotlib.pyplot as plt
import seaborn as sns
import time
import calendar

from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix, classification_report, roc_auc_score, plot_confusion_matrix
from sklearn.metrics import roc_curve, accuracy_score, precision_recall_curve, f1_score, precision_score, recall_score
from sklearn.preprocessing import StandardScaler


%matplotlib inline
sns.set_theme(palette="light:#5A9")
sns.set_context("paper", rc={"font.size":8,"axes.titlesize":8,"axes.labelsize":5})

##  Import the Datasets

We have a total of 55 csv documents which combined make up our dataset. In a first step, we will load and concatenate all of the csv files into one dataframe. 

In [56]:
import glob

In [58]:
df = pd.concat([pd.read_csv(i) for i in glob.glob("data/Kickstarter*.csv")], ignore_index=True)

In [1]:
import pandas as pd

In [59]:
df.shape

(209222, 37)

**Remark:**
Our combined Kickstarter dataset has a total of 209222 observations and 37 columns.  

## Data Cleaning

### Understanding the dataset

In [72]:
df.head(2).T

0  \
backers_count                                                           315   
blurb                                                         Babalus Shoes   
category                  {"id":266,"name":"Footwear","slug":"fashion/fo...   
converted_pledged_amount                                              28645   
country                                                                  US   
created_at                                                       1541459205   
creator                   {"id":2094277840,"name":"Lucy Conroy","slug":"...   
currency                                                                USD   
currency_symbol                                                           $   
currency_trailing_code                                                 True   
current_currency                                                        USD   
deadline                                                         1552539775   
disable_communication                                                 False   
friends                                                                 NaN   
fx_rate                                                                   1   
goal                                                                  28000   
id                                                               2108505034   
is_backing                                                              NaN   
is_starrable                                                          False   
is_starred                                                              NaN   
launched_at                                                      1548223375   
location                  {"id":2462429,"name":"Novato","slug":"novato-c...   
name                                               Babalus Children's Shoes   
permissions                                                             NaN   
photo                     {"key":"assets/023/667/205/a565fde5382d6b53276...   
pledged                                                               28645   
profile                   {"id":3508024,"project_id":3508024,"state":"in...   
slug                                                babalus-childrens-shoes   
source_url                https://www.kickstarter.com/discover/categorie...   
spotlight                                                             False   
staff_pick                                                            False   
state                                                                  live   
state_changed_at                                                 1548223375   
static_usd_rate                                                           1   
urls                      {"web":{"project":"https://www.kickstarter.com...   
usd_pledged                                                           28645   
usd_type                                                      international   

                                                                          1  
backers_count                                                            47  
blurb                     A colorful Dia de los Muertos themed oracle de...  
category                  {"id":273,"name":"Playing Cards","slug":"games...  
converted_pledged_amount                                               1950  
country                                                                  US  
created_at                                                       1501684093  
creator                   {"id":723886115,"name":"Lisa Vollrath","slug":...  
currency                                                                USD  
currency_symbol                                                           $  
currency_trailing_code                                                 True  
current_currency                                                        USD  
deadline                                                         1504976459  
disable_communication                                    

In [73]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 209222 entries, 0 to 209221
Data columns (total 37 columns):
backers_count               209222 non-null int64
blurb                       209214 non-null object
category                    209222 non-null object
converted_pledged_amount    209222 non-null int64
country                     209222 non-null object
created_at                  209222 non-null int64
creator                     209222 non-null object
currency                    209222 non-null object
currency_symbol             209222 non-null object
currency_trailing_code      209222 non-null bool
current_currency            209222 non-null object
deadline                    209222 non-null int64
disable_communication       209222 non-null bool
friends                     300 non-null object
fx_rate                     209222 non-null float64
goal                        209222 non-null float64
id                          209222 non-null int64
is_backing                  300 

**Commentary**:

Looking at the output from the commands `head()` and `info()`, there seem to a number of columns which contain information that are not usable in their current format.

In addition, `permissions`, `is_backing`, `friends` and `is_starred` only have 300 observations each. Hence, we should consider removing them. In contrast, the remaining columns are almost complete. 

Furthermore, the time related columns are not in a date format yet and therefore need to be converted. 
Finally, the columns `name`and `slug`contain the same information just in a slightly different form. We can therefore consider removing one of the columns. 

In [71]:
df.duplicated(subset='id', keep='first').sum()

26958

In [60]:
df.head(1).T

0
backers_count                                                           315
blurb                                                         Babalus Shoes
category                  {"id":266,"name":"Footwear","slug":"fashion/fo...
converted_pledged_amount                                              28645
country                                                                  US
created_at                                                       1541459205
creator                   {"id":2094277840,"name":"Lucy Conroy","slug":"...
currency                                                                USD
currency_symbol                                                           $
currency_trailing_code                                                 True
current_currency                                                        USD
deadline                                                         1552539775
disable_communication                                                 False
friends                                                                 NaN
fx_rate                                                                   1
goal                                                                  28000
id                                                               2108505034
is_backing                                                              NaN
is_starrable                                                          False
is_starred                                                              NaN
launched_at                                                      1548223375
location                  {"id":2462429,"name":"Novato","slug":"novato-c...
name                                               Babalus Children's Shoes
permissions                                                             NaN
photo                     {"key":"assets/023/667/205/a565fde5382d6b53276...
pledged                                                               28645
profile                   {"id":3508024,"project_id":3508024,"state":"in...
slug                                                babalus-childrens-shoes
source_url                https://www.kickstarter.com/discover/categorie...
spotlight                                                             False
staff_pick                                                            False
state                                                                  live
state_changed_at                                                 1548223375
static_usd_rate                                                           1
urls                      {"web":{"project":"https://www.kickstarter.com...
usd_pledged                                                           28645
usd_type                                                      international

In [45]:
df['is_starred'].unique()

array([nan, False], dtype=object)

In [41]:
df['is_backing'].unique()

array([nan, False], dtype=object)

In [42]:
df['is_starrable'].unique()

array([False,  True])

In [36]:
df["category"].unique()

array(['{"id":43,"name":"Rock","slug":"music/rock","position":17,"parent_id":14,"color":10878931,"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/music/rock"}}}',
       '{"id":54,"name":"Mixed Media","slug":"art/mixed media","position":6,"parent_id":1,"color":16760235,"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/art/mixed%20media"}}}',
       '{"id":280,"name":"Photobooks","slug":"photography/photobooks","position":5,"parent_id":15,"color":58341,"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/photography/photobooks"}}}',
       '{"id":266,"name":"Footwear","slug":"fashion/footwear","position":5,"parent_id":9,"color":16752598,"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/fashion/footwear"}}}',
       '{"id":51,"name":"Software","slug":"technology/software","position":11,"parent_id":16,"color":6526716,"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/techno

In [33]:
list(df.columns)

['backers_count',
 'blurb',
 'category',
 'converted_pledged_amount',
 'country',
 'created_at',
 'creator',
 'currency',
 'currency_symbol',
 'currency_trailing_code',
 'current_currency',
 'deadline',
 'disable_communication',
 'friends',
 'fx_rate',
 'goal',
 'id',
 'is_backing',
 'is_starrable',
 'is_starred',
 'launched_at',
 'location',
 'name',
 'permissions',
 'photo',
 'pledged',
 'profile',
 'slug',
 'source_url',
 'spotlight',
 'staff_pick',
 'state',
 'state_changed_at',
 'static_usd_rate',
 'urls',
 'usd_pledged',
 'usd_type']

In [46]:
list(df['state'].unique())

['successful', 'failed', 'live', 'canceled', 'suspended']

In [20]:
print(len(df[df['state']=="successful"]))
print(len(df[df['state']=="failed"]))
print(len(df[df['state']=="live"]))
print(len(df[df['state']=="canceled"]))
print(len(df[df['state']=="suspended"]))

2224
1276
120
149
10


In [53]:
df['usd_type'].unique()

array(['international', 'domestic', nan], dtype=object)

In [21]:
df_successful = df[df['state']=='successful']

In [39]:
df["current_currency"].unique()

array(['USD', 'CAD', 'GBP', 'AUD'], dtype=object)

In [51]:
df['permissions'].unique()

array([nan, '[]'], dtype=object)

In [25]:
df_successful.head(2).T

0  \
backers_count                                                            21   
blurb                     2006 was almost 7 years ago.... Can you believ...   
category                  {"id":43,"name":"Rock","slug":"music/rock","po...   
converted_pledged_amount                                                802   
country                                                                  US   
created_at                                                       1387659690   
creator                   {"id":1495925645,"name":"Daniel","is_registere...   
currency                                                                USD   
currency_symbol                                                           $   
currency_trailing_code                                                 True   
current_currency                                                        USD   
deadline                                                         1391899046   
disable_communication                                                 False   
friends                                                                 NaN   
fx_rate                                                                   1   
goal                                                                    200   
id                                                                287514992   
is_backing                                                              NaN   
is_starrable                                                          False   
is_starred                                                              NaN   
launched_at                                                      1388011046   
location                  {"id":2379574,"name":"Chicago","slug":"chicago...   
name                                                  New Final Round Album   
permissions                                                             NaN   
photo                     {"key":"assets/011/625/534/5bea1760d7f20943c4c...   
pledged                                                                 802   
profile                   {"id":822687,"project_id":822687,"state":"inac...   
slug                                                  new-final-round-album   
source_url                https://www.kickstarter.com/discover/categorie...   
spotlight                                                              True   
staff_pick                                                            False   
state                                                            successful   
state_changed_at                                                 1391899046   
static_usd_rate                                                           1   
urls                      {"web":{"project":"https://www.kickstarter.com...   
usd_pledged                                                             802   
usd_type                                                      international   

                                                                          1  
backers_count                                                            97  
blurb                     An adorable fantasy enamel pin series of princ...  
category                  {"id":54,"name":"Mixed Media","slug":"art/mixe...  
converted_pledged_amount                                               2259  
country                                                                  US  
created_at                                                       1549659768  
creator                   {"id":1175589980,"name":"Katherine","slug":"fr...  
currency                                                                USD  
currency_symbol                                                           $  
currency_trailing_code                                                 True  
current_currency                                                        USD  
deadline                                                         1551801611  
disable_communication                                    

In [27]:
df_failed = df[df['state']=='failed']

In [30]:
df_failed.sample(2).T

1197  \
backers_count                                                             4   
blurb                     Wir wollen ein Event veranstalten, bei dem all...   
category                  {"id":259,"name":"Civic Design","slug":"design...   
converted_pledged_amount                                                 16   
country                                                                  DE   
created_at                                                       1513432517   
creator                   {"id":210132004,"name":"Robin Gassmann","is_re...   
currency                                                                EUR   
currency_symbol                                                           €   
currency_trailing_code                                                False   
current_currency                                                        USD   
deadline                                                         1517942222   
disable_communication                                                 False   
friends                                                                 NaN   
fx_rate                                                             1.13209   
goal                                                                    100   
id                                                                 93069529   
is_backing                                                              NaN   
is_starrable                                                          False   
is_starred                                                              NaN   
launched_at                                                      1515350222   
location                  {"id":650437,"name":"Freiburg","slug":"freibur...   
name                              Open hearts for immigrant childs! - Event   
permissions                                                             NaN   
photo                     {"key":"assets/019/618/085/8c2d1bb0fb1e7af311e...   
pledged                                                                  13   
profile                   {"id":3251605,"project_id":3251605,"state":"in...   
slug                                 open-hearts-for-immigrant-childs-event   
source_url                https://www.kickstarter.com/discover/categorie...   
spotlight                                                             False   
staff_pick                                                            False   
state                                                                failed   
state_changed_at                                                 1517942222   
static_usd_rate                                                      1.2029   
urls                      {"web":{"project":"https://www.kickstarter.com...   
usd_pledged                                                         15.6377   
usd_type                                                           domestic   

                                                                       1036  
backers_count                                                             5  
blurb                     This is the start of the commercial space race...  
category                  {"id":340,"name":"Space Exploration","slug":"t...  
converted_pledged_amount                                                 50  
country                                                                  US  
created_at                                                       1498228691  
creator                   {"id":1640189118,"name":"Jeneen Price-Mills","...  
currency                                                                USD  
currency_symbol                                                           $  
currency_trailing_code                                                 True  
current_currency                                                        USD  
deadline                                                         1501089546  
disable_communication                                 

In [31]:
df_canceled = df[df['state']=='canceled']

In [32]:
df_canceled.sample(2).T

449   \
backers_count                                                             1   
blurb                     Education overcomes oppression in Treemonisha,...   
category                  {"id":24,"name":"Performance Art","slug":"art/...   
converted_pledged_amount                                                 10   
country                                                                  US   
created_at                                                       1487210294   
creator                   {"id":390369028,"name":"Miriam Miller","is_reg...   
currency                                                                USD   
currency_symbol                                                           $   
currency_trailing_code                                                 True   
current_currency                                                        USD   
deadline                                                         1489294740   
disable_communication                                                 False   
friends                                                                 NaN   
fx_rate                                                                   1   
goal                                                                   1500   
id                                                               1896708283   
is_backing                                                              NaN   
is_starrable                                                          False   
is_starred                                                              NaN   
launched_at                                                      1487964592   
location                  {"id":2355942,"name":"Arlington","slug":"arlin...   
name                      Underserved children to share seats on first c...   
permissions                                                             NaN   
photo                     {"key":"assets/015/563/776/9bf5b28e82de966e3ff...   
pledged                                                                  10   
profile                   {"id":2881064,"project_id":2881064,"state":"in...   
slug                      underserved-children-to-share-seats-on-first-c...   
source_url                https://www.kickstarter.com/discover/categorie...   
spotlight                                                             False   
staff_pick                                                            False   
state                                                              canceled   
state_changed_at                                                 1488838813   
static_usd_rate                                                           1   
urls                      {"web":{"project":"https://www.kickstarter.com...   
usd_pledged                                                              10   
usd_type                                                      international   

                                                                       1284  
backers_count                                                            10  
blurb                     Let our app - The Butler - host a party at you...  
category                  {"id":271,"name":"Live Games","slug":"games/li...  
converted_pledged_amount                                                502  
country                                                                  US  
created_at                                                       1410621005  
creator                   {"id":788704897,"name":"Moonlark Mysteries","i...  
currency                                                                USD  
currency_symbol                                                           $  
currency_trailing_code                                                 True  
current_currency                                                        USD  
deadline                                                         1414904340  
disable_communication                                 

In [47]:
df_live = df[df['state']=='live']

In [48]:
df_live.sample(3).T

1839  \
backers_count                                                             8   
blurb                     Help a small business grow and expand to reach...   
category                  {"id":10,"name":"Food","slug":"food","position...   
converted_pledged_amount                                                166   
country                                                                  AU   
created_at                                                       1548495035   
creator                   {"id":637032816,"name":"Teegan K","is_register...   
currency                                                                AUD   
currency_symbol                                                           $   
currency_trailing_code                                                 True   
current_currency                                                        USD   
deadline                                                         1553401243   
disable_communication                                                 False   
friends                                                                 NaN   
fx_rate                                                             0.70664   
goal                                                                  10000   
id                                                                975177141   
is_backing                                                              NaN   
is_starrable                                                           True   
is_starred                                                              NaN   
launched_at                                                      1550812843   
location                  {"id":1100661,"name":"Brisbane","slug":"brisba...   
name                                                         Zoe's Brownies   
permissions                                                             NaN   
photo                     {"key":"assets/024/016/292/b8360a2d948ee88c81a...   
pledged                                                              236.32   
profile                   {"id":3654242,"project_id":3654242,"state":"in...   
slug                                                          zoes-brownies   
source_url                https://www.kickstarter.com/discover/categorie...   
spotlight                                                             False   
staff_pick                                                            False   
state                                                                  live   
state_changed_at                                                 1550812843   
static_usd_rate                                                    0.716157   
urls                      {"web":{"project":"https://www.kickstarter.com...   
usd_pledged                                                         169.242   
usd_type                                                      international   

                                                                       2469  \
backers_count                                                             4   
blurb                     A Trading Card Game inspired by the Silent, Go...   
category                  {"id":34,"name":"Tabletop Games","slug":"games...   
converted_pledged_amount                                                 42   
country                                                                  US   
created_at                                                       1548834235   
creator                   {"id":2130740106,"name":"HappyPlumz","slug":"h...   
currency                                                                USD   
currency_symbol                                                           $   
currency_trailing_code                                                 True   
current_currency                                                        USD   
deadline                                                         1554336095   
disable_communication                    

In [50]:
df[df['currency']=='JPY'].sample(2).T

994   \
backers_count                                                          1705   
blurb                     We wish to bring our SRPG Visual Novel "VenusB...   
category                  {"id":35,"name":"Video Games","slug":"games/vi...   
converted_pledged_amount                                             210046   
country                                                                  JP   
created_at                                                       1530788869   
creator                   {"id":1396516721,"name":"Ninetail","is_registe...   
currency                                                                JPY   
currency_symbol                                                           ¥   
currency_trailing_code                                                False   
current_currency                                                        USD   
deadline                                                         1539995110   
disable_communication                                                 False   
friends                                                                 NaN   
fx_rate                                                          0.00896617   
goal                                                             1.6665e+07   
id                                                                596722349   
is_backing                                                              NaN   
is_starrable                                                          False   
is_starred                                                              NaN   
launched_at                                                      1537403110   
location                  {"id":1117034,"name":"Chiba-shi","slug":"chiba...   
name                       VenusBlood FRONTIER English Localization Project   
permissions                                                             NaN   
photo                     {"key":"assets/021/924/739/31b9b23fbde24e5ce48...   
pledged                                                         2.36061e+07   
profile                   {"id":3418610,"project_id":3418610,"state":"ac...   
slug                       venusblood-frontier-english-localization-project   
source_url                https://www.kickstarter.com/discover/categorie...   
spotlight                                                              True   
staff_pick                                                            False   
state                                                            successful   
state_changed_at                                                 1539995110   
static_usd_rate                                                  0.00893301   
urls                      {"web":{"project":"https://www.kickstarter.com...   
usd_pledged                                                          210874   
usd_type                                                      international   

                                                                       3007  
backers_count                                                             4  
blurb                     Using a new artificial opal "Hybrid Opal" proc...  
category                  {"id":267,"name":"Jewelry","slug":"fashion/jew...  
converted_pledged_amount                                                737  
country                                                                  JP  
created_at                                                       1508401681  
creator                   {"id":1438345805,"name":"Takeshi Hashimoto","i...  
currency                                                                JPY  
currency_symbol                                                           ¥  
currency_trailing_code                                                False  
current_currency                                                        USD  
deadline                                                         1512960824  
disable_communication                                 

In [29]:
df["spotlight"].unique()

array([ True, False])

In [19]:
import numpy as np

In [ ]:
df["state"].nu

In [6]:
df.columns

Index(['backers_count', 'blurb', 'category', 'converted_pledged_amount',
       'country', 'created_at', 'creator', 'currency', 'currency_symbol',
       'currency_trailing_code', 'current_currency', 'deadline',
       'disable_communication', 'friends', 'fx_rate', 'goal', 'id',
       'is_backing', 'is_starrable', 'is_starred', 'launched_at', 'location',
       'name', 'permissions', 'photo', 'pledged', 'profile', 'slug',
       'source_url', 'spotlight', 'staff_pick', 'state', 'state_changed_at',
       'static_usd_rate', 'urls', 'usd_pledged', 'usd_type'],
      dtype='object')

In [11]:
#df.info()

In [8]:
df1 = pd.read_csv("data/Kickstarter001.csv")

In [37]:
df1.head(2).T

0  \
backers_count                                                            30   
blurb                     Experience tea and coffee as it should be in o...   
category                  {"id":350,"name":"Pottery","slug":"crafts/pott...   
converted_pledged_amount                                               1547   
country                                                                  GB   
created_at                                                       1515610761   
creator                   {"id":1322978446,"name":"Harley Boden","is_reg...   
currency                                                                GBP   
currency_symbol                                                           £   
currency_trailing_code                                                False   
current_currency                                                        USD   
deadline                                                         1521190409   
disable_communication                                                 False   
friends                                                                 NaN   
fx_rate                                                             1.30839   
goal                                                                   1000   
id                                                               1361161119   
is_backing                                                              NaN   
is_starrable                                                          False   
is_starred                                                              NaN   
launched_at                                                      1517306009   
location                  {"id":44418,"name":"London","slug":"london-gb"...   
name                          Fine Bone China Ceramic Mugs, Made in England   
permissions                                                             NaN   
photo                     {"key":"assets/020/018/826/10db3e1b3fea41b8007...   
pledged                                                                1111   
profile                   {"id":3272613,"project_id":3272613,"state":"in...   
slug                               handcrafted-fine-bone-china-ceramic-mugs   
source_url                https://www.kickstarter.com/discover/categorie...   
spotlight                                                              True   
staff_pick                                                            False   
state                                                            successful   
state_changed_at                                                 1521190409   
static_usd_rate                                                     1.41382   
urls                      {"web":{"project":"https://www.kickstarter.com...   
usd_pledged                                                         1570.75   
usd_type                                                      international   

                                                                          1  
backers_count                                                             0  
blurb                     Playing Roles Outside of Basic Education (P.R....  
category                  {"id":49,"name":"Periodicals","slug":"publishi...  
converted_pledged_amount                                                  0  
country                                                                  US  
created_at                                                       1426362805  
creator                   {"id":1083773055,"name":"Bobby Walker","is_reg...  
currency                                                                USD  
currency_symbol                                                           $  
currency_trailing_code                                                 True  
current_currency                                                        USD  
deadline                                                         1429112946  
disable_communication                                    

In [10]:
df1.shape

(3784, 37)

In [13]:
df2=pd.read_csv("data/Kickstarter055.csv")

In [38]:
df2.head(2).T

0  \
backers_count                                                            36   
blurb                     I love drawing, and I want to share this love!...   
category                  {"id":22,"name":"Illustration","slug":"art/ill...   
converted_pledged_amount                                                465   
country                                                                  NL   
created_at                                                       1429872801   
creator                   {"id":1855992790,"name":"Nouchka","is_register...   
currency                                                                EUR   
currency_symbol                                                           €   
currency_trailing_code                                                False   
current_currency                                                        USD   
deadline                                                         1433198308   
disable_communication                                                 False   
friends                                                                 NaN   
fx_rate                                                             1.13209   
goal                                                                    150   
id                                                                323562653   
is_backing                                                              NaN   
is_starrable                                                          False   
is_starred                                                              NaN   
launched_at                                                      1430606308   
location                  {"id":733075,"name":"Rotterdam","slug":"rotter...   
name                                    Girls,girls,girls! 5 euro art sets.   
permissions                                                             NaN   
photo                     {"key":"assets/012/117/717/2c0f249e2303786501b...   
pledged                                                                 425   
profile                   {"id":1860367,"project_id":1860367,"state":"in...   
slug                                                    more-affordable-art   
source_url                https://www.kickstarter.com/discover/categorie...   
spotlight                                                              True   
staff_pick                                                            False   
state                                                            successful   
state_changed_at                                                 1433198312   
static_usd_rate                                                     1.12077   
urls                      {"web":{"project":"https://www.kickstarter.com...   
usd_pledged                                                         476.329   
usd_type                                                           domestic   

                                                                          1  
backers_count                                                            15  
blurb                     An historic event, romantic getaway, favorite ...  
category                  {"id":20,"name":"Conceptual Art","slug":"art/c...  
converted_pledged_amount                                                987  
country                                                                  US  
created_at                                                       1353894822  
creator                   {"id":2007614172,"name":"Eric Hine","is_regist...  
currency                                                                USD  
currency_symbol                                                           $  
currency_trailing_code                                                 True  
current_currency                                                        USD  
deadline                                                         1365977897  
disable_communication                                    

In [15]:
df2.shape

(965, 37)